include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,410 @@
+/**
+ * @file rotations.h
+ * @author Ethan Musser (emusser@seas.upenn.edu)
+ * @brief 
+ * @version 
+ * @date 7/8/22
+ * 
+ * @copyright 2022 The Trustees of the University of Pennsylvania. All rights reserved.
+ * 
+ */
+
+#pragma once
+
+#include <Eigen/Geometry>
+#include <type_traits>
+#include <array>
+
+namespace kodlab::rotations
+{
+
+///////////////////////////////////////////////////////////////////////////////
+// Euler Angles Representation                                               //
+///////////////////////////////////////////////////////////////////////////////
+
+enum EulerRotationType
+{
+  Extrinsic = 0,
+  Intrinsic
+};
+
+const std::array<int, 3> DEFAULT_EULER_ANGLE_AXES = {0, 1, 2};
+const EulerRotationType DEFAULT_EULER_ANGLE_ROTATION_TYPE = Extrinsic;
+
+/**
+ * @brief Euler angle rotation object.
+ * @note Default convention is <b>X-Y-Z Extrinsic</b> Euler angles.
+ *
+ * @tparam Scalar[optional] numeric type
+ */
+template<typename Scalar = float>
+struct EulerAngles
+{
+  static_assert(std::is_arithmetic<Scalar>::value,
+                "Euler angles must have a numeric type.");
+
+  /**
+   * @brief Roll angle (default radians)
+   */
+  Scalar roll = 0;
+
+  /**
+   * @brief Pitch angle (default radians)
+   */
+  Scalar pitch = 0;
+
+  /**
+   * @brief Yaw angle (default radians)
+   */
+  Scalar yaw = 0;
```

Comment:
RPY means we only get taitbryan when we act POD for other conventions 
likely should change to vector if we are going to allow that which we dont have to.

Reply:
This makes sense for generality.  However, will be documented that Tait-Bryan Extrinsic X-Y-Z is the only convention with optimized conversion functions.

Reply:
Resolved in 297e88ce81566e21f82ba644acfb587193045eed.